src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateHaFlowAction.java

```diff
@@ -58,13 +59,25 @@ protected Optional<Message> performWithResponse(State from, State to, Event even
 
             log.debug("Updating the flow {} with properties: {}", haFlowId, targetHaFlow);
 
-            // Complete target ha-flow in FSM with values from original ha-flow
+            stateMachine.setEndpointUpdateType(EndpointUpdateType.determineUpdateType(haFlow, targetHaFlow));
+
+            // Complete the target HA-flow in FSM with values from original HA-flow
             stateMachine.setTargetHaFlow(updateFlow(haFlow, targetHaFlow));
             FlowHistoryService.using(stateMachine.getCarrier()).save(HaFlowHistory
                     .of(stateMachine.getCommandContext().getCorrelationId())
                     .withAction("HA-flow properties have been updated."));
         });
 
+        if (stateMachine.getEndpointUpdateType().isPartialUpdate()) {
+            stateMachine.setNewPrimaryPathIds(stateMachine.getOldPrimaryPathIds());
+            stateMachine.setNewProtectedPathIds(stateMachine.getOldProtectedPathIds());
+
+            FlowHistoryService.using(stateMachine.getCarrier()).save(HaFlowHistory
+                    .of(stateMachine.getCommandContext().getCorrelationId())
+                    .withAction("HA-flow endpoints update is executed."));
+            stateMachine.fire(Event.UPDATE_ENDPOINTS_ONLY);
```

Comment:
FSM calls this action in state `FLOW_UPDATED`. But there is no any transition from state `FLOW_UPDATED` on event  `UPDATE_ENDPOINTS_ONLY`